Add NodeJS case

khardix opened the pull request:
Initial attempt at nodejs module switching.

These scripts install `nodegit` package dependencies, create a testing package and install `nodegit` from source as that package dependency.

From my local testing, unfortunately the only way to trigger a module-switching issue is to actually use the C dependency – `npm` itself will not complain about it being compiled against different stream.

Fixing the installation can be done by issuing `npm rebuild` in the package directory.

@junaruga, please let me know if these scripts are enough, or if you need them to do anything else.

junaruga commented:
Thanks for the PR!

> Fixing the installation can be done by issuing npm rebuild in the package directory.

That's a good information for the document. :)

> From my local testing, unfortunately the only way to trigger a module-switching issue is to actually use the C dependency

Can you put the code to "use the C dependency" in the `script/nodejs/test.sh` with a code to ignore the the command's exit status `command || true`?
It's useful to see the actual error.

junaruga commented:
I want to see `npm rebuild` is also added at the bottom of the `test.sh`.
 
```
a_command || true
npm rebuild ...
a_command # Check the command does not have an error.
```

khardix commented:
> Can you put the code to "use the C dependency" in the `script/nodejs/test.sh` with a code to ignore the the command's exit status `command || true`?
> It's useful to see the actual error.

I will try to create a quick script that would trigger this; but I have to figure out how to do that in Node, so please be patient :)

khardix commented:
> I want to see `npm rebuild` is also added at the bottom of the `test.sh`.
> 
> ```
> a_command || true
> npm rebuild ...
> a_command # Check the command does not have an error.
> ```

Added – it turns out to be easier than expected :)

junaruga commented:
https://travis-ci.org/junaruga/switch-module-stream/jobs/592599527#L6922

```
+ node -e 'require("nodegit");'
/work/nodejs-module-stream-switch-test-package/node_modules/nodegit/dist/nodegit.js:16
    throw ex;
    ^
Error: The module '/work/nodejs-module-stream-switch-test-package/node_modules/nodegit/build/Release/nodegit.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 64. This version of Node.js requires
NODE_MODULE_VERSION 72. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
...
+ npm rebuild --build-from-source
...
+ node -e 'require("nodegit");'
The command "docker run --rm -t test/fedora ${TEST_CMD}
" exited with 0.
```

Nice! It looks good to me! Thanks!
Are you ready to merge this PR?
Do you want to change the PR's subject "WIP: ", and squash it from 2 commits to 1 commit by yourself?

khardix commented:
PR renamed and squashed – feel free to merge :)

junaruga commented:
Thanks. Merged :)
